src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UpdateContactDetailsServiceTest.java

```diff
@@ -234,7 +279,241 @@ void shouldHandleRespondentDetails(String caseTypeId, Map<String, Object> caseDa
+                .nocParty(NoticeOfChangeParty.APPLICANT)
+                .applicantRepresented(YesOrNo.NO)
+                .solicitorName("Sol name")
+                .solicitorFirm("sol firm")
+                .solicitorAddress(Address.builder().addressLine1("Some Address").build())
+                .solicitorFirm("0123456789")
+                .solicitorEmail("some@email.com")
+                .solicitorAgreeToReceiveEmails(YesOrNo.YES)
+                .solicitorReference("Sol ref")
+                .build()
+            ).build();
+    }
+
+    private static FinremCaseData getContestedResponentRepresentedFinremCaseData() {
+        return FinremCaseData.builder()
+            .respondentOrganisationPolicy(OrganisationPolicy
+                .builder()
+                .organisation(Organisation
+                    .builder()
+                    .organisationID("Resp ORG ID")
+                    .organisationName("Resp ORG NAME")
+                    .build())
+                .build())
+            .respSolNotificationsEmailConsent(YesOrNo.YES)
+            .contactDetailsWrapper(ContactDetailsWrapper.builder()
+                .nocParty(NoticeOfChangeParty.RESPONDENT)
+                .contestedRespondentRepresented(YesOrNo.YES)
+                .respondentSolicitorName("Sol name")
+                .respondentSolicitorFirm("sol firm")
+                .respondentSolicitorAddress(Address.builder().addressLine1("Some Address").build())
+                .respondentSolicitorFirm("0123456789")
+                .respondentSolicitorEmail("some@email.com")
+                .respondentSolicitorDxNumber("DXnumber")
+                .build()
+            ).build();
+    }
+
+    private static FinremCaseData getConsentedResponentRepresentedFinremCaseData() {
+        return FinremCaseData.builder()
+            .respondentOrganisationPolicy(OrganisationPolicy
+                .builder()
+                .organisation(Organisation
+                    .builder()
+                    .organisationID("Resp ORG ID")
+                    .organisationName("Resp ORG NAME")
+                    .build())
+                .build())
+            .respSolNotificationsEmailConsent(YesOrNo.YES)
+            .contactDetailsWrapper(ContactDetailsWrapper.builder()
+                .nocParty(NoticeOfChangeParty.RESPONDENT)
+                .consentedRespondentRepresented(YesOrNo.YES)
+                .respondentSolicitorName("Sol name")
+                .respondentSolicitorFirm("sol firm")
+                .respondentSolicitorAddress(Address.builder().addressLine1("Some Address").build())
+                .respondentSolicitorFirm("0123456789")
+                .respondentSolicitorEmail("some@email.com")
+                .respondentSolicitorDxNumber("DXnumber")
+                .build()
+            ).build();
+    }
+
+    private static FinremCaseData getContestedNotResponentRepresentedFinremCaseData() {
```

Comment:
You can't spell respondent

Reply:
I'm going to down load the OpenDyslexic for intelij now 🤣